src/pages/index.js

```diff
@@ -56,28 +74,121 @@ const enableValidation = (config) => {
   });
 };
 enableValidation(config);
-
 /**============================================
  *               Functions
  *=============================================**/
+function handleApiCardInfo(cardData) {
+  const cardApiSection = new Section(
+    { items: cardData, renderer: renderCard },
+    cardContainer
+  );
+  cardApiSection.renderItems();
+}
 function renderCard(item, method = "append") {
-  const card = new Card(item, cardSelector, handleImageClick);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.renderCard();
   cardSection.addItem(cardElement, method);
 }
 function handleImageClick(data) {
   popupWithImage.open(data);
 }
-function handleProfileSubmit(userData) {
+function handleDeleteClick(cardData) {
+  deleteConfirmation.open(cardData);
+}
+function deleteConfirmed(card) {
+  api
+    .deleteCard(card.cardID)
+    .then(() => {
+      card.cardElement.remove();
+      card.cardElement = null;
+    })
+    .then(deleteConfirmation.close())
```

Comment:
`    .then(deleteConfirmation.close())
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 


**Please, correct it everywhere**

Reply:
I understand it better now, corrected code to wait for the response before continuing. I chose to pass unnamed arrow functions where appropriate.